protocol/routing/README.md

```diff
@@ -0,0 +1,74 @@
+IPFS Routing Protocol Spec
+==========================
+
+Authors: David Dias
+
+Reviewers:
+
+TODOS:
+
+-----------------------
+
+> This spec defines the routing protocol spec. Routing offers an interface for the features exposed by `Peer discovery` and `DHT`. The spec is a **Work In Progress**.
+
+## Supports
+
+- Routing primitives
+  - Publish and fetch content (also providing)
+- Maintaining partial state of the network
+  - DHT
+  - kbucket
+
+## Overview
+
+The Routing Protocol is composed by three componenets, these are:
+- Interface: Our routing primitives that are offered for the user, such as finding and publishing content, including the storage and lookup of metadata (Providers).
+- Peer Discovery: Responsible for filling our kbucket with best candidates.
+- Peer-to-peer Structured Overlay Network (DHT): Algorithm for the implicit network organization, based on [Coral](http://iptps03.cs.berkeley.edu/final-papers/coral.pdf) and [mainlineDHT](http://www.bittorrent.org/beps/bep_0005.html)
```

Comment:
How about this:
> - a structured overlay network: algorithms for organizing the network and for finding other nodes and data they may be hosting. (One example of such a routing network is a Kademlia DHT).

many routing protocols will not be DHT. DHT really is what our DRS is, it's just that when we say DHT today, people think this:

![](https://ipfs.io/ipfs/Qme2EsMYfMvAWiL1DfaKiFQggGdD1nviEyzYnRjs3oQ2Tq/cap.png)

and not this:

![](https://ipfs.io/ipfs/QmNcc5DNcrLzHUeSGNUEfcNYPb3e6oga2JsK47KM9oBz7J/database.jpg)